File: implementations/go/agent/agent_test.go
package agent

import "testing"

func TestAgentEventTypesIncludesAll6RegistryEntries(t *testing.T) {
	expected := []string{
		"agent.message.sent",
		"agent.message.received",
		"agent.message.failed",
		"agent.request.created",
		"agent.response.created",
		"agent.decision.recorded",
	}
	for _, typ := range expected {
		if !EventTypes[typ] {
			t.Errorf("missing: %s", typ)
		}
	}
	if len(EventTypes) != len(expected) {
		t.Errorf("expected %d types, got %d", len(expected), len(EventTypes))
	}
}

func TestIsEventTypePositives(t *testing.T) {
	for _, typ := range []string{
		"agent.message.sent",
		"agent.message.received",
		"agent.message.failed",
		"agent.request.created",
		"agent.response.created",
		"agent.decision.recorded",
	} {
		if !IsEventType(typ) {
			t.Errorf("expected true for %s", typ)
		}
	}
}

func TestIsEventTypeNegatives(t *testing.T) {
	for _, typ := range []string{
		"task.submitted",
		"session.opened",
		"context.updated",
		"",
	} {
		if IsEventType(typ) {
			t.Errorf("expected false for %s", typ)
		}
	}
}

File: implementations/typescript/packages/agent/test/agent.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { AGENT_EVENT_TYPES, isAgentEventType } from "../src/index.js";

test("agent event types includes all 6 registry entries", () => {
  const expected = [
    "agent.message.sent",
    "agent.message.received",
    "agent.message.failed",
    "agent.request.created",
    "agent.response.created",
    "agent.decision.recorded"
  ];
  for (const type of expected) {
    assert.equal(AGENT_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(AGENT_EVENT_TYPES.size, expected.length);
});

test("isAgentEventType positives", () => {
  assert.equal(isAgentEventType("agent.message.sent"), true);
  assert.equal(isAgentEventType("agent.message.received"), true);
  assert.equal(isAgentEventType("agent.message.failed"), true);
  assert.equal(isAgentEventType("agent.request.created"), true);
  assert.equal(isAgentEventType("agent.response.created"), true);
  assert.equal(isAgentEventType("agent.decision.recorded"), true);
});

test("isAgentEventType negatives", () => {
  assert.equal(isAgentEventType("task.submitted"), false);
  assert.equal(isAgentEventType("session.opened"), false);
  assert.equal(isAgentEventType("context.updated"), false);
  assert.equal(isAgentEventType(""), false);
});
